engine/Shopware/Components/Translation.php

```diff
@@ -36,11 +36,48 @@ class Shopware_Components_Translation
     private $connection;
 
     /**
+     * @var int
+     */
+    private $language;
+
+    /**
+     * @var int
+     */
+    private $fallback;
+
+    /**
      * @param Connection|null $connection
+     * @param Container|null  $container
      */
-    public function __construct(Connection $connection = null)
+    public function __construct(Connection $connection = null, $container = null)
     {
-        $this->connection = $connection ?: Shopware()->Container()->get('dbal_connection');
+        $container = ($container) ?: Shopware()->Container();
+        $this->connection = $connection ?: $container->get('dbal_connection');
+
+        // Determine how to query the current language
+        if ($container->has('Auth')) {
+            $locale = $identity = Shopware()->Container()->get('Auth')->getIdentity()->locale;
+
+            if ($locale) {
+                $this->language = $locale->getId();
+            }
+        } elseif ($container->has('Shop')) {
+            $localeName = $container->get('Shop')->getLocale()->getLocale();
+            $locale = Shopware()->Models()->getRepository('Shopware\Models\Shop\Locale')->findOneBy([
```

Comment:
Please use `$container` instead of `Shopware()->Models()`.

Reply:
Also, why do we query the locale here again. We already have it as an object in the shop, don't we?